### ISSUE TITLE
Posting a string instead of an object

### ISSUE THREAD

**Issue by jordanpowell88** [paulcbetts/refit#329]
I am trying to post to an api end point that is expecting a string in the payload instead of an object.  I'm not sure if it's something I'm doing wrong or a bug. Here is what I have

**Interface**
```
[Post("api/endpoint/id")]
[Headers("Content-Type: application/json")]
Task<HttpResponse> PostComment(string id, [Body] string comment);
```

**Controller**
`var postComment = await api.PostComment(id, viewModel.Comment);`

I checked and my viewModel property is set correctly before posting and I am getting a successful status code as well but the viewModel.Comment isn't being passed correctly into the interface string comment.

Any ideas?

**Comment by stephannielsen** [comment]
I am trying to do the same. I am also in control of the server side. From what I see is that the API (asp.net WebApi 2) is expecting the string with quotes around it, but refit is sending the string without them.
If I send a string with escaped quotes (e.g. "\"my value\""), the Api accepts the request.

I think the Api side handles this correctly. Using quotes around the string does make sense.

So how can I tell Refit to format the value correctly?
I am using Refit 2.4.1.

**Comment by bennor** [comment]
Refit doesn't support posting a string body as a JSON string automatically. Instead, when you post a string value as the body, [the assumption is that _want to_ send the raw string](https://github.com/paulcbetts/refit/blob/master/Refit/RequestBuilderImplementation.cs#L104) i.e. you have already serialized the request body.

If we were to change this, it would be a breaking change for anyone using that feature.

What I would suggest (given you control the API) is to change your API to accept an object:

```
// Web API
IHttpActionResult SomeApi(SomeRequest request) {
  string value = request.Value;
  // Implementation
}

// Refit
Task<SomeResponse> PostSomething([Body] SomeRequest request);

await api.PostSomething(new SomeRequest{ Value = "foo" });
```

If it's a common enough thing in your API, you could even use a generic `StringRequest` type.

**Comment by bennor** [comment]
It's referenced here in the docs: https://github.com/paulcbetts/refit#body-content

**Comment by stephannielsen** [comment]
Thank you for the quick answer.

I understand that this would be a breaking change. But having some option here to customize the behaviour doesn't imply a breaking change. After all I do specify the content type as `application/json` for the endpoint explicitly, and it is also possible to specify the BodySerializationMethod as JSON.

A simple if/else inside the if that transforms the string to `StringContent` could respect the setting. In default, Refit would not change from existing behaviour.

**Comment by bennor** [comment]
That's where it gets tricky though. The default value for serialization format when just `[Body]` is specified _is_ JSON, and since there is no explicit "raw" serialization method on the enumeration, it would be a breaking change for anyone using the default behaviour for a string body parameter.

Might be clearer with an example. If you actually are passing a string body and want it sent as-is, you'll have:

```csharp
Task SendTheThing([Body] string thing);
```

Which is effectively identical to what you're suggesting, as there's an implicit `BodySerializationMethod.Json` on that attribute, and we're ignoring it for strings.

**Comment by onovotny** [comment]
The workaround for this would be to use a `StringContent` as the parameter type in the interface and add the value there.

**Comment by onovotny** [comment]
We can set the implicit value of `BodySerializationMethod` to `BodySerializationMethod.None` and treat that behavior as we do now. Then if `BodySerializationMethod.Json` is explicitly set, we can json-ify the input string.

Wonder if that'd work....

**Comment by bennor** [comment]
Not sure why this didn't autoclose - should have been closed by #425
